src/main/java/io/percy/appium/providers/AppAutomate.java

```diff
@@ -162,11 +163,13 @@ public String screenshot(String name, ScreenshotOptions options) {
 
         super.setDebugUrl(getDebugUrl(result));
         try {
-            percyScreenshotUrl = super.screenshot(name, options, osVersion, device);
+            JSONObject response = super.screenshot(name, options, osVersion, device);
+            percyScreenshotUrl = response.getString("link");
+            return response;
         } catch (Exception e) {
             error = e.getMessage();
         }
-        executePercyScreenshotEnd(name, percyScreenshotUrl, error);
+        executePercyScreenshotEnd(name, percyScreenshotUrl, error, options.getSync());
```

Comment:
no need to pass in end call

Reply:
added this for instrumentation in seleniumHub. There is a selenium hub pr going there.